lib/ace/mode/latex_parser.js

```diff
@@ -33,26 +62,81 @@ var LatexParser = exports.LatexParser = function(bgTokenizer) {
+                        doSpellcheck && parseRowForSpellcheck.call(this, token.value, row, column);
+                        break;
+                    case "text":
+                        doGetDiscussions && saveLastReadDiscussionComment.call(this);
+                        doSpellcheck && parseRowForSpellcheck.call(this, token.value, row, column);
                         break;
                     default:
-                        // Skip tokens we don't want to spellcheck
+                        doGetDiscussions && saveLastReadDiscussionComment.call(this);
+                        // skip tokens we don't want to parse
                         break;
                 }
             }
         }
+        // save last read comment in case of the document ends with this comment
+        doGetDiscussions && saveLastReadDiscussionComment.call(this);
+
+        function parseRowForSpellcheck(tokenValue, row, column) {
+            var words = this.splitTextToWords(tokenValue, row, column);
+            for (var i = 0, wordsLength = words.length; i < wordsLength; ++i) {
+                if (!this.myDictionary.check(words[i].value)) {
+                    this.myErrors.push({
+                        row: words[i].row,
+                        column: words[i].column,
+                        text: "grammar error",
+                        type: "error",
+                        raw: words[i].value
+                    });
+                }
+            }
+        }
+
+        function parseRowForDiscussions(tokenValue, row) {
+            // comment format: % TYPE <AUTHOR> DATETIME:
+            // {
+            //     TYPE: '*' for the first comment in discussion, '^' for replies
+            //     AUTHOR: any string, e.g. email
+            //     DATETIME: any string
+            // }
+            var matchComment = tokenValue.match(DISCUSSION_COMMENT_REGEXP);
+            if (matchComment) {
+                // if we met a new discussion title, stop reading body of the previous discussion
+                // and start a new one
+                saveLastReadDiscussionComment.call(this);
+                this.myLastReadDiscussionComment = {
+                    type: matchComment[1],
+                    author: matchComment[2],
+                    datetime: matchComment[4],
+                    row: row,
+                    text: []
+                };
+            } else if (this.myLastReadDiscussionComment !== null) {
+                var matchTextRow = tokenValue.match(DISCUSSION_TEXT_ROW_REGEXP);
+                if (matchTextRow) {
+                    this.myLastReadDiscussionComment.text.push(matchTextRow[1]);
+                }
+            }
+        }
+
+        function saveLastReadDiscussionComment() {
+            if (this.myLastReadDiscussionComment !== null) {
+                if (this.myLastReadDiscussionComment.type === '*') {
+                    this.myLastReadDiscussionComment.replies = [];
```

Comment:
why don't you init this property when you started parsing this comment (at line 110)?